src/app/new-char/page.tsx

```diff
@@ -0,0 +1,120 @@
+    reader.onload = () => {
+      const newCharacter: pjData = {
+        name: formData.get('characterName')! as string,
+        image: reader.result as string,
+        profession: formData.get('characterProfession')! as string,
+        color: formData.get('characterColor')! as string,
+        life: parseInt(formData.get('characterLife')! as string),
+        mana: parseInt(formData.get('characterMana')! as string),
+        currentLife: parseInt(formData.get('characterLife')! as string),
+        currentMana: parseInt(formData.get('characterMana')! as string),
+        level: parseInt(formData.get('characterLevel')! as string),
+        armorClass: parseInt(formData.get('characterAc')! as string),
+      };
+
+      if (players === undefined)
+        return alert('Erro 500 algo de errado aconteceu...');
+      setPlayers([...players, newCharacter]);
+    };
+  };
+
+  return (
+    <Wrapper>
+      <Form action={sendToServer}>
+        <Wrapper>
+          <label>Character Avatar:</label>
+          <input
+            required={true}
+            name='characterAvatar'
+            accept='.jpg, .png'
+            placeholder='Character Avatar'
+            type='file'
+          />
+        </Wrapper>
+        <Wrapper>
+          <label>Character Name:</label>
+          <input
+            required={true}
+            name='characterName'
+            placeholder='Name'
+            type='text'
+          />
+        </Wrapper>
+        <Wrapper>
+          <label>Character Level:</label>
+          <input
+            required={true}
+            name='characterLevel'
+            placeholder='Level'
+            type='number'
+          />
+        </Wrapper>
+        <Wrapper>
+          <label>Character Color:</label>
+          <input
+            required={true}
+            name='characterColor'
+            placeholder='Color'
+            type='color'
+          />
+        </Wrapper>
+        <Wrapper>
+          <label>Profession Name:</label>
+          <input
+            required={true}
+            name='characterProfession'
+            placeholder='Profession'
+            type='text'
+          />
+        </Wrapper>
+        <Wrapper>
+          <label>Life</label>
+          <input
+            defaultValue={0}
+            name='characterLife'
+            placeholder='Life'
+            type='number'
+          />
+        </Wrapper>
+        <Wrapper>
+          <label>Mana</label>
+          <input
+            defaultValue={0}
+            name='characterMana'
+            placeholder='Mana'
+            type='number'
+          />
+        </Wrapper>
+        <Wrapper>
+          <label>Armor Class</label>
+          <input
+            defaultValue={0}
+            name='characterAc'
+            placeholder='Armor Class'
+            type='number'
+          />
+        </Wrapper>
+        <Button color='toggle' type='submit'>
+          Create character!
+        </Button>
```

Comment:
Adicionar uma _prop_ `disabled` pro botão e utilizar um _useState_ para bloquear no instante o usuário der o _submit_ e reabilitar somente quando o char for criado ou der erro

Reply:
Acho que foi resolvido no commit 6de33bce4e85a10716e280b8004c575a3c4b78ca